README.md
# TEP NextJS 

![version](https://img.shields.io/badge/version-1.1.0-blue.svg)

### Project Description


TEP is Free and Open Source. It features multiple HTML and NextJS elements and it comes with dynamic components for NextJS.

This project aims to satisfy an academic Project proposed in the higher institute of applied science and technologies of sousse [ISSATSo](http://www.issatso.rnu.tn/fo/index.php).

This project is build with both presentation pages, and Profiles.


### About NextJs Components

TEP NextJS is built with over frontend 120 components, giving you the freedom of choosing and combining. All components can take variations in colors, that you can easily modify using Tailwind CSS classes (NOTE: each time you add a new class, a class that does not exist in `src/assets/styles/tailwind.css`, you will need to compile again tailwind.

Every element has multiple states for colors, styles, hover, focus, that you can easily access and use.


### CSS Components

TEP NextJS comes with 120 Fully Coded CSS elements, such as [Alerts](), [Buttons](), [Inputs]() and many more.

## 🧾 Contents

* [Project tree](#project-tree)
* [Browser Support](#browser-support)
* [Reporting Issues](#reporting-issues)
* [Getting Started](#getting-started)
    * [Installing](#installing)
    * [Running the project](#running-the-project)
* [Authors](#authors) 


## 🌳 Project tree

```
TEP-nextjs
.
├── CHANGELOG.md
├── LICENSE.md
├── README.md
├── public
│   ├── img
│   │   ├── brand
│   │   │   └── favicon.ico
│   │   ├── github.svg
│   │   ├── LinkedIn.svg
│   │   └── google.svg
│   └── styles
│       ├── index.css
│       └── tailwind.css
├── components
│   ├── Cards
│   │   ├── CardProfile.js
│   │   └── CardSettings.js
│   ├── Dropdowns
│   │   └── UserDropdown.js
│   ├── Footers
│   │   ├── Footer.js
│   │   ├── FooterAdmin.js
│   │   └── FooterSmall.js
│   ├── Headers
│   │   └── HeaderStats.js
│   ├── Maps
│   │   └── MapExample.js
│   ├── Navbars
│   │   ├── AdminNavbar.js
│   │   ├── AuthNavbar.js
│   │   └── IndexNavbar.js
│   ├── PageChange
│   │   └── PageChange.js
│   ├── SEOs
│   │   └── SEO.js
│   └── Sidebar
│       └── Sidebar.js
├── layouts
│   ├── Admin.js
│   └── Auth.js
├── package.json
├── package-lock.json
├── pages
│   ├── 404.js
│   ├── _app.js
│   ├── _document.js
│   ├── _error.js
│   ├── admin
│   │   ├── landing.js
│   │   ├── profile.js
│   │   └── settings.js
│   ├── auth
│   │   ├── login.js
│   │   └── register.js
│   └──index.js
└── tailwind.config.js
```

## 📋 Browser Support

At present, we officially aim to support the last two versions of the following browsers:

| Chrome | Firefox | Edge | Safari | Opera |
|:---:|:---:|:---:|:---:|:---:|
| <img src="https://github.com/creativetimofficial/public-assets/blob/master/logos/chrome-logo.png?raw=true" width="64" height="64"> | <img src="https://raw.githubusercontent.com/creativetimofficial/public-assets/master/logos/firefox-logo.png" width="64" height="64"> | <img src="https://raw.githubusercontent.com/creativetimofficial/public-assets/master/logos/edge-logo.png" width="64" height="64"> | <img src="https://raw.githubusercontent.com/creativetimofficial/public-assets/master/logos/safari-logo.png" width="64" height="64"> | <img src="https://raw.githubusercontent.com/creativetimofficial/public-assets/master/logos/opera-logo.png" width="64" height="64"> |

## 🤔 Reporting Issues

We use GitHub Issues as the official bug tracker for the TEP NextJS. Here are some advices for our users that want to report an issue:

1. Make sure that you are using the latest version of NextJS.
2. Providing us reproducible steps for the issue will shorten the time it takes for it to be fixed.
3. Some issues may be browser specific, so specifying in what browser you encountered the issue might help.



# 🏁 Getting Started
## 📥 Installing
Node version must be 16 or higher.
```sh
> npm install
```
or
```sh
> yarn install
```

## 🌠 Running the project
Execute this command to run the project:

```sh
> npm run dev
```
## 💼 Authors
* [**BOUZIR Mohamed Ali**](https://www.linkedin.com/in/bouzir-mohamed-ali/) - *Developer*
* [**BEN MABROUK Ons**](https://www.linkedin.com/in/ons-ben-mabrouk-499904197/) - *Developer*


CHANGELOG.md
### Major style changes
- The upgrade of Tailwind CSS from version 1 to version 2, will cause multiple style changes, check them out on the official Tailwind CSS websites:
  - https://blog.tailwindcss.com/tailwindcss-v2
  - https://tailwindcss.com/
  - https://tailwindcss.com/docs/upgrading-to-v2
### Deleted components
- `next.config.js` (we do not need it anymore since we now use `Built-In CSS Support` from `NextJS`)
  - for the absolute imports, we've used the `jsconfig.json` file
### Added components
- `jsconfig.json` (to keep our absolute imports)
### Deleted dependencies
- `@tailwindcss/custom-forms`
- `react-google-maps` (replaced by simple Google Maps API)
- `@types/googlemaps` (dependencies of `react-google-maps`)
- `@types/markerclustererplus` (dependencies of `react-google-maps`)
- `@types/react` (dependencies of `react-google-maps`)
- `@zeit/next-css` (we'll use the default `Built-In CSS Support` from `NextJS`)
- `@zeit/next-sass` (we'll use the default `Built-In CSS Support` from `NextJS`)
- `node-sass` (we'll use the default `Built-In CSS Support` from `NextJS`)
- `next-images` (we'll use the default `Built-In CSS Support` from `NextJS`)
- `next-fonts` (we'll use the default `Built-In CSS Support` from `NextJS`)
- `next-compose-plugins` (we'll use the default `Built-In CSS Support` from `NextJS`)
- `path` (we'll use the default `Built-In CSS Support` from `NextJS`)
- `webpack` (we'll use the default `Built-In CSS Support` from `NextJS`)
### Added dependencies
- `@tailwindcss/forms` (replaces `@tailwindcss/custom-forms`)
- `autoprefixer`
- `postcss`
### Updated dependencies
```
@fortawesome/fontawesome-free    5.14.0   →   5.15.3
@popperjs/core                    2.5.1   →    2.9.1
chart.js                          2.9.3   →    2.9.4
next                              9.5.3   →   10.0.9
react                           16.13.1   →   17.0.1
@types/react                    16.9.49   →   17.0.3
react-dom                       16.13.1   →   17.0.1
react-scripts                     3.4.3   →    4.0.3
tailwindcss                      1.8.10   →    2.0.4
typescript                        4.0.3   →    4.2.3
```
### Warning
_On a clean install there may be some warnings from request, chokidar, fsevents - they come from node_modules, and they do not affect the product at all._
### Original Release
- Added design from Tailwind Starter Kit by Creative Tim


pages/admin/landing.js
import React from "react";
import Link from "next/link";
import { useState, useEffect } from "react";

// components

import Navbar from "components/Navbars/AdminNavbar.js";
import Footer from "components/Footers/Footer.js";

export default function Landing() {

  const [messages, setMessages] = useState([""]);
  const [addedMessage, setAddedMessage] = useState(false);
  
  useEffect(() => {
    setTimeout(() => {
      fetch('http://localhost:8081/getAllMessages',  {Authorization : localStorage.getItem('JWTtoken')}
      )
          .then(response => {
            if(!response.ok) {
              throw Error('could not fetch tge data for that ressource :( ')
            }
            return response.json()})
          .then(data => {
            setMessages(data);
            })
          .catch (err => {
            console.log(err.message)
          })
        }, 1000)
setAddedMessage(false)
});

const createMessage = async (event) => {
  event.preventDefault();
  // Get data from the form.
  const data = {
      title : event.target.title.value,
      content : event.target.content.value
}
           fetch('http://localhost:8080/api/messages/new' , { method : 'POST',
           headers : {
             'Content-Type': 'application/json',
             'Authorization' : localStorage.getItem('JWTtoken')
           },
           body : JSON.stringify(data)})
          
        .then(response => {
            if(!response.ok) {
              throw Error('could not fetch the data for that ressource :( ')
            }else{
              setAddedMessage(true);
            }
            return response.json()})

          .catch (err => {
            console.log(err.message)
          })
          document.getElementById("create-message-form").reset();
}

const handleLike = (id) => {
  fetch('http://localhost:8080/api/messages/'+id+'/vote/like' , { method : 'POST',
  headers : {
    'Content-Type': 'application/json',
    'Authorization' : localStorage.getItem('JWTtoken')
  }})
 
.then(response => {
   if(!response.ok) {
     throw Error('could not fetch the data for that ressource :( ')
   }
   return response.json()})

 .catch (err => {
   console.log(err.message)
 })
}

const handleDislike = (id) => {
  fetch('http://localhost:8080/api/messages/'+id+'/vote/dislike' , { method : 'POST',
  headers : {
    'Content-Type': 'application/json',
    'Authorization' : localStorage.getItem('JWTtoken')
  }})
 
.then(response => {
   if(!response.ok) {
     throw Error('could not fetch the data for that ressource :( ')
   }
   return response.json()})

 .catch (err => {
   console.log(err.message)
 })
}
  return (
    <>
      <Navbar transparent />
      <main>
        <div className="relative pt-16 pb-32 flex content-center items-center justify-center min-h-screen-75">
          <div
            className="absolute top-0 w-full h-full bg-center bg-cover"
            style={{
              backgroundImage:
                "url('https://images.unsplash.com/photo-1557804506-669a67965ba0?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1267&q=80')",
            }}
          >
            <span
              id="blackOverlay"
              className="w-full h-full absolute opacity-75 bg-black"
            ></span>
          </div>
          <div className="container relative mx-auto">
            <div className="items-center flex flex-wrap">
              <div className="w-full lg:w-6/12 px-4 ml-auto mr-auto text-center">
                <div className="pr-12">
                  <h1 className="text-white font-semibold text-5xl">
                    Giant News-Wall
                  </h1>
                  <p className="mt-4 text-lg text-blueGray-200">
                    Over here we will be watching all peoples new posts
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            className="top-auto bottom-0 left-0 right-0 w-full absolute pointer-events-none overflow-hidden h-16"
            style={{ transform: "translateZ(0)" }}
          >
            <svg
              className="absolute bottom-0 overflow-hidden"
              xmlns="http://www.w3.org/2000/svg"
              preserveAspectRatio="none"
              version="1.1"
              viewBox="0 0 2560 100"
              x="0"
              y="0"
            >
              <polygon
                className="text-blueGray-200 fill-current"
                points="2560 0 2560 100 0 100"
              ></polygon>
            </svg>
          </div>
        </div>

        <section className="pb-20 bg-blueGray-200 -mt-24">
          <div className="container mx-auto px-4">
            <div className="flex flex-wrap">
              <div className="lg:pt-12 pt-6 w-full md:w-4/12 px-4 text-center">
                
              </div>

              <div className="w-full md:w-12 px-6 text-center">
                <div className="relative flex flex-col min-w-0 break-words bg-white w-full mb-8 shadow-lg rounded-lg">
                  <div className="px-4 py-5 flex-auto">
                    <div className="text-white p-3 text-center inline-flex items-center justify-center w-12 h-12 mb-5 shadow-lg rounded-full bg-lightBlue-400">
                      <i className="fas fa-retweet"></i>
                    </div>
                    <h6 className="text-xl font-semibold">What's new mate</h6>
                    <form onSubmit={createMessage} id="create-message-form">
                    <div className="relative w-full mb-3">
                    <label
                      className="block uppercase text-blueGray-600 text-xs font-bold mb-2"
                      htmlFor="grid-Message"
                    >
                      Message Title
                    </label>
                    <input
                    name="title"
                      type="text"
                      className="border-0 px-3 py-3 placeholder-blueGray-300 text-blueGray-600 bg-white rounded text-sm shadow focus:outline-none focus:ring w-full ease-linear transition-all duration-150"
                      placeholder="exemple: My wonderfull title"
                    />
                  </div>

                  <div className="relative w-full mb-3">
                    <label
                   
                      className="block uppercase text-blueGray-600 text-xs font-bold mb-2"
                      htmlFor="grid-Message"
                    >
                      Message
                    </label>
                    <input
                    name="content"
                      type="text"
                      className="border-0 px-3 py-6 placeholder-blueGray-300 text-blueGray-600 bg-white rounded text-sm shadow focus:outline-none focus:ring w-full ease-linear transition-all duration-150"
                      placeholder="How are you doing today ? tell us😊"
                    />
                  </div>
                  
                  <div className="text-center mt-6">
                    <button
                      className="bg-blueGray-800 text-white active:bg-blueGray-600 text-sm font-bold uppercase px-6 py-3 rounded shadow hover:shadow-lg outline-none focus:outline-none mr-1 mb-1 w-full ease-linear transition-all duration-150"
                      type="submit"
                    >
                      Submit
                    </button>
                  </div>
                </form>
                  </div>
                </div>
              </div>
            </div>

            
          </div>
        </section>

        
        <section className="pb-20 relative block bg-blueGray-800">
        <div
          className="bottom-auto top-0 left-0 right-0 w-full absolute pointer-events-none overflow-hidden -mt-20 h-20"
          style={{ transform: "translateZ(0)" }}
        >
          <svg
            className="absolute bottom-0 overflow-hidden"
            xmlns="http://www.w3.org/2000/svg"
            preserveAspectRatio="none"
            version="1.1"
            viewBox="0 0 2560 100"
            x="0"
            y="0"
          >
            <polygon
              className="text-blueGray-800 fill-current"
              points="2560 0 2560 100 0 100"
            ></polygon>
          </svg>
        </div>

        <div className="container mx-auto px-4 lg:pt-24 lg:pb-64">
          <div className="flex flex-wrap text-center justify-center">
            <div className="w-full lg:w-6/12 px-4">
              <h2 className="text-4xl font-semibold text-white">
                All Posts
              </h2>
            </div>
          </div>
        </div>
      </section>
      <section className="relative block py-24 lg:pt-0 bg-blueGray-800">
        <div className="container mx-auto px-4">
          <div className="flex flex-wrap justify-center lg:-mt-64 -mt-48">
            <div className="w-full lg:w-8/12 px-4">
              {messages.map((message, id) => {
                return (
                  <div key={id} className="relative flex flex-col min-w-0 break-words w-full mb-6 shadow-lg rounded-lg bg-blueGray-200">
                  <div className="flex-auto p-5 lg:p-10">
                    {/*Waiting for mapping */}
                    <h4 className="text-2xl font-semibold">
                    {message.title}
                    </h4>
                    <p className="leading-relaxed mt-1 mb-4 text-blueGray-500">
                    {message.content}
                    </p>
                   
                    <div className="relative w-full mb-3">
                    <a
                    className="bg-lightBlue-600  text-white w-8 h-8 rounded-full outline-none focus:outline-none mr-1 mb-1 inline-flex items-center justify-center"
                    onClick={() => handleLike(message.id)}
                    type="button"
                  >
                    <i className="fa fa-thumbs-up" style={{color: 'white'}}></i>
                  </a>
                  <p className=" w-8 h-8 mr-1 mb-1 inline-flex  items-center justify-center">{message.likes}</p>
                  <a
                  className="bg-lightBlue-600  text-white w-8 h-8 rounded-full outline-none focus:outline-none mr-1 mb-1 inline-flex items-center justify-center"
                  onClick={() => handleDislike(message.id)}
                  type="button"
                >
                  <i className="fa fa-thumbs-down" style={{color: 'white'}}></i>
                </a>
                  </div>
                    </div>
                    </div>
                );
              })}
            </div>
          </div>
        </div>
      </section>


        {/*Implementors */}
        <section className="pt-20 pb-48">
          <div className="container mx-auto px-4">
            <div className="flex flex-wrap justify-center text-center mb-24">
              <div className="w-full lg:w-6/12 px-4">
                <h2 className="text-4xl font-semibold">This Web Application Founders</h2>
                <p className="text-lg leading-relaxed m-4 text-blueGray-500">
                  A highly motivated developers worked on developping this web application.
                </p>
              </div>
            </div>
            <div className="flex flex-wrap">
              <div className="w-full md:w-6/12 lg:w-3/12 lg:mb-0 mb-12 px-4">
                <div className="px-6">
                </div>
              </div>
              <div className="w-full md:w-6/12 lg:w-3/12 lg:mb-0 mb-12 px-4">
                <div className="px-6">
                  <img
                    alt="..."
                    src="https://lh3.googleusercontent.com/a-/AOh14GgMa0QtGyROGStkcqwD4C4s_MKAF12u8-q6V_NC=s83-c-mo"
                    className="shadow-lg rounded-full mx-auto max-w-120-px"
                  />
                  <div className="pt-6 text-center">
                    <h5 className="text-xl font-bold">Ben Mabrouk Ons</h5>
                    <p className="mt-1 text-sm text-blueGray-400 uppercase font-semibold">
                      Full Stack <br></br> Web Developer
                    </p>
                    <div className="mt-6">
                      <a
                        className="bg-lightBlue-600  text-white w-8 h-8 rounded-full outline-none focus:outline-none mr-1 mb-1 inline-flex items-center justify-center"
                        href="https://www.facebook.com/ons.benmabrouk.372"
                        target={"_blank"}
                      >
                        <i className="fab fa-facebook-f"></i>
                      </a>
                      <a
                        className="bg-lightBlue-400  text-white w-8 h-8 rounded-full outline-none focus:outline-none mr-1 mb-1 inline-flex items-center justify-center"
                        href="https://www.linkedin.com/in/ons-ben-mabrouk-499904197/"
                        target={"_blank"}
                      >
                        <i className="fab fa-linkedin"></i>
                      </a>
                      <a
                        className="bg-pink-500  text-white w-8 h-8 rounded-full outline-none focus:outline-none mr-1 mb-1 inline-flex items-center justify-center"
                        href="https://www.instagram.com/ons.benmabrouk/"
                        target={"_blank"}
                      >
                        <i className="fab fa-instagram"></i>
                      </a>
                      <a
                        className="bg-black  text-white w-8 h-8 rounded-full outline-none focus:outline-none mr-1 mb-1 inline-flex items-center justify-center"
                        href="https://github.com/Ons-bm"
                        target={"_blank"}
                      >
                        <i className="fab fa-github"></i>
                      </a>
                    </div>
                  </div>
                </div>
              </div>
              <div className="w-full md:w-6/12 lg:w-3/12 lg:mb-0 mb-12 px-4">
                <div className="px-6">
                  <img
                  alt="..."
                  src="https://lh3.googleusercontent.com/a-/AOh14Gh_XOXTapsAGuPpOuTdFmTM1egSmFdESWgZouYJkg=s83-c-mo"
                  className="shadow-lg rounded-full mx-auto max-w-120-px"
                />
                  <div className="pt-6 text-center">
                    <h5 className="text-xl font-bold">Bouzir Mohamed Ali</h5>
                    <p className="mt-1 text-sm text-blueGray-400 uppercase font-semibold">
                      Full Stack Js <br></br> Web Developer
                    </p>
                    <div className="mt-6">
                        <a
                        className="bg-lightBlue-600  text-white w-8 h-8 rounded-full outline-none focus:outline-none mr-1 mb-1 inline-flex items-center justify-center"
                        href="https://www.facebook.com/Daly.Bzr/"
                        target={"_blank"}
                      >
                        <i className="fab fa-facebook-f"></i>
                      </a>
                      <a
                        className="bg-lightBlue-400  text-white w-8 h-8 rounded-full outline-none focus:outline-none mr-1 mb-1 inline-flex items-center justify-center"
                        href="https://www.linkedin.com/in/bouzir-mohamed-ali/"
                        target={"_blank"}
                      >
                        <i className="fab fa-linkedin"></i>
                      </a>
                      <a
                        className="bg-pink-500  text-white w-8 h-8 rounded-full outline-none focus:outline-none mr-1 mb-1 inline-flex items-center justify-center"
                        href="https://www.instagram.com/med_ali_bz/"
                        target={"_blank"}
                      >
                        <i className="fab fa-instagram"></i>
                      </a>
                      <a
                        className="bg-black  text-white w-8 h-8 rounded-full outline-none focus:outline-none mr-1 mb-1 inline-flex items-center justify-center"
                        href="https://github.com/MohamedAliBouzir"
                        target={"_blank"}
                      >
                        <i className="fab fa-github"></i>
                      </a>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </section>

        
      </main>
      <Footer />
    </>
  );
}
